### ISSUE TITLE
Incorrect Inferred type for DDPMScheduler.from_pretrained model_id

### ISSUE THREAD

**Issue by qnlbnsl** [huggingface/diffusers#3799]
### Describe the bug

When using DDPMScheduler.from_pretrained to load up the noise scheduler  the param pretrained_model_name_or_path is set to type Dict[str, Any] [https://github.com/huggingface/diffusers/blob/afcca3916fafccff1419c45f3de00be57db12dfc/src/diffusers/schedulers/scheduling_utils.py#LL80C62-L80C62](url) whereas in the documentation at [https://huggingface.co/docs/diffusers/v0.17.1/en/api/schedulers/overview#diffusers.SchedulerMixin.from_pretrained.pretrained_model_name_or_path](url) it is stated that it is a string.

BTTW could just be my inexperience with python

### Reproduction

use pylance and initialize the ddpmsceduler.from_pretrained

### Logs

```shell
pylance:
(method) def from_pretrained(
    pretrained_model_name_or_path: Dict[str, Any] = None,
    subfolder: str | None = None,
    return_unused_kwargs: bool = False,
    **kwargs: Unknown
) -> Unknown
Instantiate a Scheduler class from a pre-defined JSON configuration file inside a directory or Hub repo.

Parameters:
    pretrained_model_name_or_path (str or os.PathLike, *optional*):
        Can be either:

  - A string, the \*model id\* of a model repo on huggingface.co.
----------------------------------------------------------------------------------------
Error:
Argument of type "str" cannot be assigned to parameter "pretrained_model_name_or_path" of type "Dict[str, Any]" in function "from_pretrained"
  "str" is incompatible with "Dict[str, Any]"PylancereportGeneralTypeIssues
```

### System Info

- `diffusers` version: 0.16.1
- Platform: Linux-5.19.0-1025-gcp-x86_64-with-glibc2.35
- Python version: 3.10.6
- PyTorch version (GPU?): 2.0.0+cu117 (True)
- Huggingface_hub version: 0.15.1
- Transformers version: 4.25.1
- Accelerate version: 0.19.0
- xFormers version: 0.0.18
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: not sure.. just running on one gpu within a docker container

**Comment by qnlbnsl** [comment]
Sure! here is a little snippet. Make sure to turn strict type checks on using pydantic

```python
from diffusers import (
    DDPMScheduler,
)

pretrained_model_name_or_path="Some path to a pretrained model"

noise_scheduler = DDPMScheduler.from_pretrained( # Type check fails here for the model name
    pretrained_model_name_or_path, subfolder="scheduler"
)
```

**Comment by patrickvonplaten** [comment]
Not 100% sure what the error is here. The following should work:

```py
from diffusers import (
    DDPMScheduler,
)

pretrained_model_name_or_path="runwayml/stable-diffusion-v1-5"

noise_scheduler = DDPMScheduler.from_pretrained( # Type check fails here for the model name
    pretrained_model_name_or_path, subfolder="scheduler"
)
```

**Comment by qnlbnsl** [comment]
The issue is the incorrect type in the declaration. It works,, yes, but only when you ignore the type. This is not a critical bug or anything just something that diverges from the docs.
![image](https://github.com/huggingface/diffusers/assets/46204848/b17a17d0-dc7b-400a-ab06-6823655cd1f3)
